src/frontend/src/catalog/system_catalog/rw_catalog/rw_sinks.rs

```diff
@@ -61,7 +63,12 @@ fn read_rw_sinks_info(reader: &SysCatalogReaderImpl) -> Result<Vec<RwSink>> {
                     .to_uppercase(),
                 sink_type: sink.sink_type.to_proto().as_str_name().into(),
                 connection_id: sink.connection_id.map(|id| id.connection_id() as i32),
-                definition: sink.create_sql(),
+                definition: if sink.owner.user_id == user_id {
```

Comment:
I'm wondering if this (only showing raw SQL if the user is the owner) is a common behavior in other DBMS? Even so, I guess it would be based on the permission system but not simply determined by the ownership. cc @yezizp2012 

In my personal opinion I would suggest adding a new column of `redacted_definition`. Users are allowed to decide which form to reveal or share with us.

Reply:
> adding a new column of redacted_definition
+1 for source/table/sink system tables.

What about show create? Another `SHOW REDACTD CREATE`? That can be done in another PR.